hs_core/views/apps.py

```diff
@@ -0,0 +1,28 @@
+from django.views.generic import TemplateView
+
+from hs_tools_resource.models import ToolResource
+
+APPROVED_APPLICATIONS = [
+    694,    # SWATShare
+    679,    # Hydroshare GIS
+    5316,   # JupyterHub NCSA
+    826,    # CUAHSI Data Series Viewer
+    2640    # National Water Model Forecast viewer
+]
+
```

Comment:
@aphelionz I am thinking these resource ids can be probably retrieved automatically without hardcoding them in this constant list. Are those app types such as "SWATShare", "Hydrshare GIS", etc. stored in ToolResource metadata somewhere so that they can be compared against to represent set of approved apps by types? If not, probably we should add it rather than hardcoding these ids here or am I missing something?

Reply:
I'm ok with these being hardcoded since they are for "CUAHSI-approved" apps. Otherwise we would need a system where only admins can edit certain types of metadata, or to add a new model and admin page for this. New app approvals are infrequent enough that I'm OK with this until it becomes a problem.

Reply:
@aphelionz OK. Then that is fine with me. My last comment related to this is that whether it is easier to hardcode web app tool resource 32 bit UUIDs rather than the resource ids. I am thinking to find out approved web app tool resource UUIDs, admin can just use HS discovery to search resources by type to find the tool resource that needs to go into the approval list, then add the UUID into this hardcoded approval list. However, I assume to find the resource ID for a specific resource, you'll have to look into the database or use a django command to get it, which would take at least one extra not-so-straightforward step. What do you think?

Reply:
It may be better to get the list of approved apps using a class method of ToolResource (```ToolResource.get_approved_apps()```). For now this should return the apps based on the hard coded app ids. Later the logic for approved apps can be changed in this class method.

Reply:
ok, I can move the app ids to that function and use the UUIDS, no problem